e2e/testcases/workload_identity_test.go

```diff
@@ -598,8 +583,27 @@ func updateRSyncWithGitSourceConfig(nt *nomostest.NT, rs client.Object, repo *gi
 						}
 					}
 				}`, sc.dir))
-	rsRef := client.ObjectKey{Name: rs.GetName(), Namespace: rs.GetNamespace()}
-	return rsyncValidateMeta{rsRef: rsRef, sha1Func: sc.commitFn, syncDir: sc.dir}
+	id, err := lookupObjectID(rs, nt.Scheme)
+	if err != nil {
+		nt.T.Fatalf("invalid object ID: %v", err)
+	}
+	return rsyncValidateMeta{rsID: id, sha1Func: sc.commitFn, syncDir: sc.dir}
+}
+
+// TODO: promote to helper function, or find existing alternative
+func lookupObjectID(obj client.Object, scheme *runtime.Scheme) (core.ID, error) {
```

Comment:
What about using `core.IDOf`? It doesn't verify the scheme, but should work for RSync objects.

Reply:
IDOf only works if the GK is set on the object, which isn't true of all objects. That convention is unique to Config Sync code and the k8sobjects libs, but it's not shared by upstream k8s code or client-go.

I found a similar lookup function in my fake client code, so I'm going to move it into the core package next to IDof, so it's easy to discover.

At some point in the future, we should require everything to check the scheme unless the object is Unstructured, but that would be a big refactor to handle the possible errors.

Reply:
Had to put it in kinds.LookupID to avoid a package dependency cycle